diff -up ratpoints-2.2.1/Makefile.orig ratpoints-2.2.1/Makefile
--- ratpoints-2.2.1/Makefile.orig	2022-09-26 19:35:32.265005341 +0200
+++ ratpoints-2.2.1/Makefile	2022-09-26 19:42:39.289489826 +0200
@@ -22,6 +22,7 @@
 #   Michael Stoll, September 21, 2009; January 7, 2022
 #   with changes by Bill Allombert, December 29, 2021
 
+MAJOR = 0
 PRIME_SIZE = 7
 VERSION = 2.2.1
 
@@ -29,9 +30,13 @@ CC = gcc
 RM = rm -f
 INSTALL = cp
 
-INSTALL_DIR = /usr/local
+PREFIX = /usr
+BINDIR = /usr/bin
+INCDIR = /usr/include
+LIBDIR = /usr/lib
+INSTALL = install
 
-CCFLAGS0 = -Wall -O2 -fomit-frame-pointer -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE}
+CCFLAGS0 = -Wall -O2 -fomit-frame-pointer -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE} -fPIC
 # For gcc on Apple, may have to add '-fnested-functions' to CCFLAGS0.
 # Add "-DUSE_LONG_IN_PHASE_2" to work with unsigned long's instead of bit-arrays
 #  in phase 2 of the sieving. This is usually slower.
@@ -84,11 +89,11 @@ TEMPFILES = sift.o init.o sturm.o find_p
             rptest.out sift-debug.o find_points-debug.o main.o test2.out
 
 # Executables and library produced when building
-TARGETFILES = ratpoints libratpoints.a rptest ratpoints-debug ratpoints-doc-2.2.pdf
+TARGETFILES = ratpoints libratpoints.so.$(MAJOR) rptest ratpoints-debug ratpoints-doc-2.2.pdf
 
 FAILED = "Test failed!"
 
-all: ratpoints libratpoints.a doc
+all: ratpoints libratpoints.so.$(MAJOR) doc
 
 doc: ratpoints-doc-2.2.pdf
 
@@ -113,14 +118,13 @@ timing: ratpoints
 	time ./ratpoints '1 0 126 0 441' 400000 -q > /dev/null
 
 install-bin: ratpoints
-	${INSTALL} ratpoints ${INSTALL_DIR}/bin/
-	chmod 755 ${INSTALL_DIR}/bin/ratpoints
+	${INSTALL} -D - m755 ratpoints ${DESTDIR}/${BINDIR}/ratpoints
 
-install-lib: ratpoints.h libratpoints.a
-	${INSTALL} ratpoints.h ${INSTALL_DIR}/include/
-	chmod 644 ${INSTALL_DIR}/include/ratpoints.h
-	${INSTALL} libratpoints.a ${INSTALL_DIR}/lib/
-	chmod 644 ${INSTALL_DIR}/lib/libratpoints.a
+install-lib: ratpoints.h libratpoints.$(MAJOR)
+	${INSTALL} -D -m644 ratpoints.h ${DESTDIR}${INCDIR}/ratpoints.h
+	${INSTALL} -d ${DESTDIR}${LIBDIR}
+	${INSTALL} -m755 libratpoints.so.$(MAJOR) ${DESTDIR}${LIBDIR}
+	ln -s libratpoints.so.$(MAJOR) ${DESTDIR}${LIBDIR}/libratpoints.so
 
 install: install-bin install-lib
 
@@ -145,10 +149,11 @@ distclean: clean
 
 debug: ratpoints-debug
 
-libratpoints.a: sift.o init.o sturm.o find_points.o
-	ar rs libratpoints.a sift.o init.o sturm.o find_points.o
+libratpoints.so.$(MAJOR): sift.o init.o sturm.o find_points.o
+	${CC} -shared -o libratpoints.so.$(MAJOR) -Wl,-soname=libratpoints.so.$(MAJOR) sift.o init.o sturm.o find_points.o
+	ln -s libratpoints.so.$(MAJOR) libratpoints.so
 
-ratpoints: libratpoints.a main.c ratpoints.h
+ratpoints: libratpoints.so.$(MAJOR) main.c ratpoints.h
 	${CC} main.c -o ratpoints ${CCFLAGS_0} ${CCFLAGS2} ${CCFLAGS3} ${CCFLAGS}
 
 main.o: main.c ratpoints.h
@@ -185,7 +190,7 @@ find_points.o: find_points.c ratpoints.h
 find_points-debug.o: find_points.c ratpoints.h rp-private.h primes.h find_points.h
 	${CC} find_points.c -c -o find_points-debug.o ${CCFLAGS_0} -DDEBUG ${CCFLAGS}
 
-rptest: libratpoints.a rptest.c ratpoints.h testdata.h
+rptest: libratpoints.so.$(MAJOR) rptest.c ratpoints.h testdata.h
 	${CC} rptest.c -o rptest ${CCFLAGS_0} ${CCFLAGS2} ${CCFLAGS3} ${CCFLAGS}
 
 gen_init_sieve_h: gen_init_sieve_h.c ratpoints.h rp-private.h primes.h
